src/app/core/components/header/header.component.ts

```diff
@@ -1,61 +1,67 @@
-import { Component, OnInit } from '@angular/core'
+import { Component, OnDestroy, OnInit } from '@angular/core'
 import { Event, NavigationEnd, Router, RouterEvent } from '@angular/router'
+import { Subscription } from 'rxjs'
+import { filter } from 'rxjs/operators'
 import { LoginService } from 'src/app/auth/services/login.service'
 import { SearchService } from 'src/app/youtube/services/search.service'
 import { SortbarManagerService } from 'src/app/youtube/services/sortbar-manager.service'
-import { filter } from 'rxjs/operators'
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss'],
+  styleUrls: ['./header.component.scss']
 })
-export class HeaderComponent implements OnInit {
-  // @Output() changeSortVisibility = new EventEmitter<boolean>()
+export class HeaderComponent implements OnInit, OnDestroy {
+  isSearchPage = false
 
-  // @Output() search = new EventEmitter()
+  userSubscription!: Subscription
 
-  isSearchPage!: boolean
+  query = ''
+
+  isUserLogged!: boolean
```

Comment:
if u can avoid using ! operator, do so. Assign default value instead

Reply:
suddenly, this field already has a default value in my local version, smth strange...